client/src/components/LecturePage.js

```diff
@@ -18,7 +18,19 @@ function LecturePage(props) {
             .then((b) => {
                 setBookings(b);
             }).catch((e) => { setState(() => { throw e; }) })
-    }, []);
+    }, [props.lecture_id]);
+
+    let cancelLecture = async () => {
+        if(lecture){
+            API.cancelLecture(lecture.id)
+            .then((result)=>{
+                API.getLecture(lecture.id)
+                .then((l) => {
+                    setLecture(l);
+                }).catch((e) => { setState(() => { throw e; }) })
+            }).catch((e) => { setState(() => { throw e; }) })
+        }
+    }
```

Comment:
Since this is an async function, we could use await inside it, and avoid having a `then...catch` chain

Reply:
It is necessary to throw again the error, for some reason it does not work if you don't throw it again.
We can test removing it, but I remember it gave me problems